zcash_primitives/src/consensus.rs

```diff
@@ -67,6 +69,10 @@ pub enum NetworkUpgrade {
     ///
     /// [Heartwood]: https://z.cash/upgrade/heartwood/
     Heartwood,
+    /// The [Canopy] network upgrade.
+    ///
+    /// [Canopy]: https://z.cash/upgrade/canopy/
+    Canopy,
```

Comment:
The placeholders added here will conflict with master both for adding Canopy there and updating Heartwood. But we can address that when we eventually rebase this branch on a more recent base.